osu.Game.Tournament/Screens/Editors/TournamentEditorScreen.cs

```diff
@@ -17,19 +20,42 @@
 
 namespace osu.Game.Tournament.Screens.Editors
 {
-    public abstract class TournamentEditorScreen<TDrawable, TModel> : TournamentScreen, IProvideVideo
+    public abstract class TournamentEditorScreen<TDrawable, TModel> : TournamentScreen, IProvideVideo, IKeyBindingHandler<GlobalAction>
         where TDrawable : Drawable, IModelBacked<TModel>
         where TModel : class, new()
     {
         protected abstract BindableList<TModel> Storage { get; }
 
         private FillFlowContainer<TDrawable> flow;
 
+        [Resolved(canBeNull: true)]
+        private TournamentSceneManager sceneManager { get; set; }
+
         protected ControlPanel ControlPanel;
 
+        protected virtual bool IsSubScreen => false;
+
+        protected virtual System.Type ParentScreen { get; set; }
```

Comment:
Rather than having this virtual, how about having the subscreen take a nullable parent in its constructor. so changing `TournamentEditorScreen` take an optional `OsuScreen` and storing that reference. That was we wouldn't have to resort to `System.Type` here (it can be an `OsuScreen`), although for the display call we will need to use `.GetType()` still.